### PR TITLE
Remove default unspecified authentication context class from AuthRequest creation

### PR DESCRIPTION
**Pull request by gabalese**
Explicitly pass a list of auth_context_classes in order to create a `RequestedAuthContext` element in the SAML auth request.

The support for different authentication context classes, and the semantics around the different classes may be interpreted differently and may potentially cause interoperability problems with systems such as ADFS 3.0.

### CODE REVIEW

**Comment by obmarg** [comment]
These SAML changes could do with a lot more background.  What _is_ a RequestedAuthnContext? 

What issues did you encountering that led to this change?

"May be interepreted differently" - in what way?

"May potentially cause interoperability problems" - what kinds of problems?

File: setup.py [hunk 1]
```diff
@@ -42,10 +43,10 @@ def run(self):
 
 setup(
     name='onelogin.saml',
-    version='0.0.1',
+    version='0.0.2',
```
Comment (villaaston) [inline setup.py]: This doesn't seem like a patch version, the behaviour for adding `auth_context_classes` has completely changed now there's no default

Comment (obmarg) [inline setup.py]: Don't think we should be upticking version numbers as part of feature/fix PRs either...

Comment (gabalese) [inline setup.py]: Valid point. This looks more like a 0.1.0 release to me.
Do you mean we should bump the version number in a separate commit/pr, or that we shouldn't use a `fix/etc` PR if we're making relevant changes?

File: onelogin/saml/AuthRequest.py [hunk 2]
```diff
@@ -51,7 +51,7 @@ def create(_clock=None, _uuid=None, _zlib=None, _base64=None,
     issuer = kwargs.pop('issuer')
     name_identifier_format = kwargs.pop('name_identifier_format')
     idp_sso_target_url = kwargs.pop('idp_sso_target_url')
-    auth_context_classes = kwargs.pop('auth_context_classes', ['unspecified'])
+    auth_context_classes = kwargs.get('auth_context_classes')
```
Comment (gabalese) [inline onelogin/saml/AuthRequest.py]: Not that I can see, unless we want to add a flag somewhere to say something like `RequestedAuthnContext=False`.

Comment (gabalese) [inline onelogin/saml/AuthRequest.py]: Also, it's worth noting that _theoretically_ not providing any context is the real way to provide an unspecified context.

**Comment by gabalese** [comment]
🔁

**Comment by gabalese** [comment]
🔁 , please?

**Comment by obmarg** [comment]
🔀

**Pull request merged**